### ISSUE TITLE
plotObservedVsSimulated() - error when observed data time exceeds simulation time

### ISSUE THREAD

**Issue by PavelBal** [Open-Systems-Pharmacology/OSPSuite-R#1075]
[SimResults.csv](https://github.com/Open-Systems-Pharmacology/OSPSuite-R/files/9310133/SimResults.csv)

With the following simulated and observed data:

```
simFilePath <- system.file("extdata", "Aciclovir.pkml", package = "ospsuite")
sim <- loadSimulation(simFilePath)

simData <- importResultsFromCSV(sim, "d:/SimResults.csv")

obsData <- DataSet$new(name = "Observed")
obsData$setValues(xValues = c(1, 3, 3.5, 4, 5), yValues = c(1.9, 6.1, 7, 8.2, 1))
obsData$xUnit <- "min"

myDC <- DataCombined$new()
myDC$addSimulationResults(simData, groups = "myGroup")
myDC$addDataSets(obsData, groups = "myGroup")

plotIndividualTimeProfile(myDC)
```

![image](https://user-images.githubusercontent.com/25903040/184180280-3b4a1c30-176f-4b46-bd09-e9b0dbd02b42.png)

`plotObservedVsSimulated()` fails:

```
> plotObservedVsSimulated(myDC)
Error in rep(seq(1, base - 1, length.out = ticks_per_base), reps) : 
  invalid 'times' argument
In addition: There were 11 warnings (use warnings() to see them)
```

**Comment by PavelBal** [comment]
Last point should be ignored, as specified in https://github.com/Open-Systems-Pharmacology/OSPSuite-R/issues/804

**Comment by IndrajeetPatil** [comment]
This bug needs to be resolved in _tlf_ first:

https://github.com/Open-Systems-Pharmacology/TLF-Library/issues/367

**Comment by IndrajeetPatil** [comment]
Reproducible example:

``` r
library(ospsuite)
#> Loading required package: rClr
#> Loading the dynamic library for Microsoft .NET runtime...
#> Loaded Common Language Runtime version 4.0.30319.42000
options(tibble.width = Inf)

simFilePath <- system.file("extdata", "Aciclovir.pkml", package = "ospsuite")
sim <- loadSimulation(simFilePath)

simData <- withr::with_tempdir({
  df <- dplyr::tibble(
    IndividualId = c(0, 0, 0),
    `Time [min]` = c(0, 2, 4),
    `Organism|PeripheralVenousBlood|Aciclovir|Plasma (Peripheral Venous Blood) [µmol/l]` = c(0, 4, 8)
  )
  readr::write_csv(df, "SimResults.csv")
  importResultsFromCSV(sim, "SimResults.csv")
})

obsData <- DataSet$new(name = "Observed")
obsData$setValues(xValues = c(1, 3, 3.5, 4, 5), yValues = c(1.9, 6.1, 7, 8.2, 1))
obsData$xUnit <- "min"

myDC <- DataCombined$new()
myDC$addSimulationResults(simData, groups = "myGroup")
myDC$addDataSets(obsData, groups = "myGroup")

plotObservedVsSimulated(myDC)
#> Warning: Removed 4 rows containing missing values (geom_segment).
#> Removed 4 rows containing missing values (geom_segment).
```

![](https://i.imgur.com/tUM04E0.png)

<sup>Created on 2022-08-16 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1.9000)</sup>

**Comment by IndrajeetPatil** [comment]
This should be fixed by #1090 

``` r
library(ospsuite)
#> Loading required package: rClr
#> Loading the dynamic library for Microsoft .NET runtime...
#> Loaded Common Language Runtime version 4.0.30319.42000

simFilePath <- system.file("extdata", "Aciclovir.pkml", package = "ospsuite")
sim <- loadSimulation(simFilePath)

simData <- withr::with_tempdir({
  df <- dplyr::tibble(
    IndividualId = c(0, 0, 0),
    `Time [min]` = c(0, 2, 4),
    `Organism|PeripheralVenousBlood|Aciclovir|Plasma (Peripheral Venous Blood) [µmol/l]` = c(0, 4, 8)
  )
  readr::write_csv(df, "SimResults.csv")
  importResultsFromCSV(sim, "SimResults.csv")
})

obsData <- DataSet$new(name = "Observed")
obsData$setValues(xValues = c(1, 3, 3.5, 4, 5), yValues = c(1.9, 6.1, 7, 8.2, 1))
obsData$xUnit <- "min"

myDC <- DataCombined$new()
myDC$addSimulationResults(simData, groups = "myGroup")
myDC$addDataSets(obsData, groups = "myGroup")

plotObservedVsSimulated(myDC)
#> Predicted values couldn't be interpolated at following time points:
#> 5
#> Warning in plotObservedVsSimulated(myDC):
#> Warning: Removed 4 rows containing missing values (geom_segment).
#> Removed 4 rows containing missing values (geom_segment).
```

![](https://i.imgur.com/i1wfAjy.png)

<sup>Created on 2022-08-16 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1.9000)</sup>
